api/lib/application/authentication/oidc/oidc-controller.js

```diff
@@ -12,4 +15,28 @@ module.exports = {
 
     return h.response({ redirectLogoutUrl }).code(200);
   },
+
+  async findUserForOidcReconciliation(request) {
+    const authenticationKey = request.query['authentication-key'];
+    const { email, password, identityProviderCode } = accountReconciliationSerializer.deserialize(request.payload);
+
+    const oidcAuthenticationService = authenticationRegistry.lookupAuthenticationService(identityProviderCode);
+
+    const result = await usecases.findUserForOidcReconciliation({
+      email,
+      password,
+      identityProviderCode,
+      authenticationKey,
+      oidcAuthenticationService,
+    });
+
+    if (result.isAuthenticationComplete) {
+      return {
+        access_token: result.pixAccessToken,
+        logout_url_uuid: result.logoutUrlUUID,
+      };
+    } else {
+      return { authentication_key: result.authenticationKey };
+    }
```

Comment:
```suggestion
    }
    
    return { authentication_key: result.authenticationKey };
```

On se pose la question du retour de l'authenticationKey qui à ce niveau existe déjà dans le front et le controller pourrait retourner un code HTTP 204

Reply:
C'est vrai qu'on l'a déjà. Ça va dépendre de comment on gère ça coté front, si ça être complexe de faire le tri entre la 204, la 200 et l'objet retournée